### ISSUE TITLE
[ChatGPT QA] Face Detect & Crop returns the full source image when no face is detected

### ISSUE THREAD

**Issue by xiaden** [xiaden/comfyui-imgutils#3]
## Finding
`Imgutils Bbox Crop` converts a zero-result detection into one successful-looking crop containing the entire source image. This breaks the shipped `Face Detect & Crop` subgraph: when face detection legitimately finds no faces, the subgraph outputs the full input image on its `faces` output instead of representing that no face crops exist.

The same fallback also turns malformed or otherwise non-usable bbox JSON into the original image, conflating "no valid boxes" with "one box covering the whole image." Downstream workflows therefore cannot distinguish a real crop from detection failure/absence by output cardinality.

## User/System Journey
1. A user runs the repository-provided `Face Detect & Crop` subgraph on a supported image.
2. The subgraph routes the image through `ImgUtilsDetect` in `Detect Faces` mode at confidence `0.5`.
3. For an image with no face above threshold, `ImgUtilsDetect` returns JSON `[]`.
4. The subgraph passes that JSON directly into `ImgUtilsBboxCrop`, as defined in `subgraphs/Face Detect & Crop.json`.
5. `ImgUtilsBboxCrop.execute()` produces no crops, enters `if not crops`, and returns `[pil_to_comfy(pil)]` — the complete source image.
6. The subgraph exposes that image as its `faces` output, so a downstream face-only workflow receives one ordinary IMAGE item even though zero faces were detected.

The same supported Detect/OCR → Bbox Crop journey reaches the fallback when a high confidence threshold yields zero detections. No injected state or direct internal caller is required.

## Files in scope
- `utility/node_bbox_crop.py` — `ImgUtilsBboxCrop.execute()` maps zero parsed/valid boxes to a one-element list containing the original image despite its documented "one cropped image per bbox" contract.
- `detect/node_detect.py` — normal zero-detection path emits `[]`, which is a valid upstream value consumed by Bbox Crop.
- `detect/node_ocr.py` — another supported producer that can legitimately emit an empty bbox array.
- `subgraphs/Face Detect & Crop.json` — repository-provided end-to-end workflow wires Detect JSON directly to Bbox Crop and publishes the result as `faces`.
- `README.md` — documents Bbox Crop as returning one image per Detect/OCR bbox.

## Expected behavior
A valid zero-detection result should preserve the semantic distinction between zero detected regions and one detected region. The `Face Detect & Crop` journey must not label or emit the complete source frame as a detected face merely because no bbox was produced.

Malformed bbox input should likewise fail clearly or produce the node's defined empty-result behavior rather than silently becoming a successful full-image crop.

## Actual behavior
When parsing yields zero valid bboxes, `ImgUtilsBboxCrop` returns a list containing the original full image. Consequently, a no-face result from the supported `Face Detect & Crop` subgraph has output cardinality 1 and contains the untouched source frame. Any downstream consumer that iterates, saves, embeds, classifies, or otherwise treats `faces` as detected face crops operates on false-positive data.

## Correction direction
Restore the cardinality/meaning contract at the bbox-crop boundary: zero valid boxes must not be represented as one full-image crop. Define an explicit empty/no-region behavior compatible with ComfyUI list outputs (or fail clearly if the framework cannot represent an empty IMAGE list), and keep malformed-input handling distinguishable from a genuine whole-image bbox. Add an end-to-end regression around the shipped `Face Detect & Crop` subgraph or its exact Detect → Bbox Crop path proving that a legitimate `[]` detection cannot surface the source image as a face crop.

## Metadata
- QA-Agent: ChatGPT
- Reviewed-SHA: `f88cd73d17a549281dce4149da0d47c7e09ab870`
- Branch: `main`
- Severity: High
- Category: correctness
- Review-Mode: user-journey-head
- QA-Fingerprint: `xiaden/comfyui-imgutils|correctness|ImgUtilsBboxCrop.execute|zero-valid-bboxes-fallback-to-original-image`

**Comment by xiaden** [comment]
### Additional remediation-scope evidence: Bbox Mask has the same zero-cardinality contract break

The same supported Detect/OCR → bbox utility journey also reaches `ImgUtilsBboxMask.execute()` in `utility/node_bbox_mask.py`. When upstream detection legitimately emits `[]`, or parsing produces no usable boxes, Bbox Mask returns a **one-element mask list** containing an all-zero full-frame mask:

```python
if not masks:
    empty = torch.zeros((1, h, w), dtype=torch.float32)
    return io.NodeOutput([empty])
```

README documents Bbox Mask as returning “one mask per bbox,” so zero bboxes should not become one mask item. This is less visually dangerous than Bbox Crop returning the source image, but it breaks the same cardinality invariant: downstream list iteration sees one detected region when the detector produced zero regions.

This should be included in #3's correction rather than split into another issue: define and regression-test a consistent zero-region list-output contract across both `ImgUtilsBboxCrop` and `ImgUtilsBboxMask`, including the normal Detect/OCR `[]` path.
